frontend/src/components/dashboard/settings/custom-domain/CustomDomain.tsx

```diff
@@ -0,0 +1,130 @@
+  const [customFromAddresses, setCustomFromAddresses] = useState([] as string[])
+  const modalContext = useContext(ModalContext)
+  const { email } = useContext(AuthContext)
+  const title = 'From Address'
+
+  async function onVerifyFromAddressClicked(label: string) {
+    sendUserEvent(GA_USER_EVENTS.ADD_FROM_ADDRESS)
+    modalContext.setModalContent(
+      <VerifyCustomDomainModal
+        label={label}
+        onSuccess={onSuccess}
+      ></VerifyCustomDomainModal>
+    )
+  }
+
+  async function populateFromAddresses() {
+    const fromAddresses = await getCustomFromAddresses()
+    setCustomFromAddresses(fromAddresses)
+  }
+
+  // Get custom from addresses
+  useEffect(() => {
+    populateFromAddresses()
+  }, [])
+
+  function renderFromAddresses() {
+    return (
+      <>
+        <p>You can now send emails from the following email addresses: </p>
+        <table className={styles.credTable}>
+          <tbody>
+            <tr>
+              <th className="lg">Label</th>
+              <th></th>
+            </tr>
+            {customFromAddresses.map((label: string) => (
+              <tr key={label}>
+                <td className="lg">{label}</td>
+                <td className={cx('sm', styles.actionColumn)}>
+                  <i
+                    className={cx(
+                      'bx',
+                      'bx-list-check',
+                      styles.icon,
+                      styles.verifyButton
+                    )}
+                    onClick={() => onVerifyFromAddressClicked(label)}
+                  ></i>
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </>
+    )
+  }
+  function renderRequestForm() {
+    return (
+      <>
+        <p>
+          You can now add your email address <b>{email}</b> as the From Address.
+          Recipients will receive emails from this address, instead of the
+          default{' '}
+          <b>
+            <Trans id="defaultMailFrom" />
+          </b>
+          .
+        </p>
+        <p>
+          The process requires correspondence with your agency&apos;s IT team,
+          and is manual at the moment, so please follow the steps below.
+        </p>
+        <p>
+          <b>How to proceed</b>:
+        </p>
+        <ol>
+          <li>Fill in request form on FormSG</li>
+          <li>Wait for us to contact you within 5 working days</li>
+        </ol>
+
+        <div className={styles.actionButtons}>
+          <OutboundLink
+            eventLabel={i18n._(LINKS.customDomainRequestUrl)}
+            to={i18n._(LINKS.customDomainRequestUrl)}
+            target="_blank"
+          >
+            <PrimaryButton className={styles.request}>
+              Fill in request form <i className="bx bx-right-arrow-alt"></i>
+            </PrimaryButton>
+          </OutboundLink>
+          <PrimaryButton
+            className={styles.status}
+            onClick={() => onVerifyFromAddressClicked(email)}
```

Comment:
Quick check, will users be allowed to customise the display name of their From Address?

Reply:
I didn't add an input box for that, but the backend supports a custom display name.

Reply:
Added the input box